frontstage/error_handlers.py

```diff
@@ -47,3 +50,12 @@ def connection_error_jwt_validation(error):  # pylint: disable=unused-argument
     return redirect(url_for('error_bp.not_logged_in_error_page',
                             _external=True,
                             _scheme=getenv('SCHEME', 'http')))
+
+
+@app.errorhandler(RequestEntityTooLarge)
+def request_entity_too_large_error(error):
+    case_id = request.args.get('case_id', None)
+    logger.error('Upload failed', status_code=413, case_id=case_id)
+    error_info = {'status code': 413, 'text': 'The spreadsheet must be smaller than 20MB in size.'}
+    return render_template('surveys/surveys-upload-failure.html', _theme='default', error_info=error_info,
+                           case_id=case_id)
```

Comment:
We should probably test this as well but could potentially go in without

Reply:
And even thought this is a failure we may want to restrict access to this page to authorised users